diff --git a/Demo/Application/Features/PayPalWebCheckoutViewController.swift b/Demo/Application/Features/PayPalWebCheckoutViewController.swift
index f42490775..f44130189 100644
--- a/Demo/Application/Features/PayPalWebCheckoutViewController.swift
+++ b/Demo/Application/Features/PayPalWebCheckoutViewController.swift
@@ -97,17 +97,41 @@ class PayPalWebCheckoutViewController: PaymentButtonBaseViewController {
         request.lineItems = [lineItem]        
         request.offerPayLater = payLaterToggle.isOn
         request.intent = newPayPalCheckoutToggle.isOn ? .sale : .authorize
-
-        payPalClient.tokenize(request) { nonce, error in
-            sender.isEnabled = true
-
-            guard let nonce else {
-                self.progressBlock(error?.localizedDescription)
+        
+        // Flow 1 - Potential (likely) merchant integration pattern to trigger quick sequence of `v1/config` calls.
+        
+        BraintreeDemoMerchantAPIClient.shared.createCustomerAndFetchClientToken { clientToken, error in
+            guard let clientToken else {
+                self.progressBlock("Error fetching Client Token")
                 return
             }
+            
+            let freshAPIClient = BTAPIClient(authorization: clientToken)!
+            let newPayPalClient = BTPayPalClient(apiClient: freshAPIClient)
+            newPayPalClient.tokenize(request) { nonce, error in
+                sender.isEnabled = true
 
-            self.completionBlock(nonce)
+                guard let nonce else {
+                    self.progressBlock(error?.localizedDescription)
+                    return
+                }
+
+                self.completionBlock(nonce)
+            }
         }
+
+        // Flow 2 - Original Demo app setup. Doesn't trigger the issue as easily.
+        
+//        payPalClient.tokenize(request) { nonce, error in
+//            sender.isEnabled = true
+//
+//            guard let nonce else {
+//                self.progressBlock(error?.localizedDescription)
+//                return
+//            }
+//
+//            self.completionBlock(nonce)
+//        }
     }
     
     // MARK: - Vault Flows
diff --git a/Sources/BraintreeCore/BTHTTP.swift b/Sources/BraintreeCore/BTHTTP.swift
index 1cc046507..d702a5e19 100644
--- a/Sources/BraintreeCore/BTHTTP.swift
+++ b/Sources/BraintreeCore/BTHTTP.swift
@@ -398,6 +398,8 @@ class BTHTTP: NSObject, URLSessionTaskDelegate {
                     path = mutationName
                 }
                 
+                print("🔥Path: \(path)")
+                
                 networkTimingDelegate?.fetchAPITiming(
                     path: path,
                     connectionStartTime: transaction.connectStartDate?.utcTimestampMilliseconds,
